demos/rain_prediction.ipynb

```diff
@@ -0,0 +1,467 @@
+    "\n",
+    "#Register the commit hook to be executed after a PUT_FILE event\n",
+    "fs.register_hook(FSEvent.PUT_FILE, commit_on_put)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When you execute the next cell however, you will see a message indicating that the upload of the resource has been skipped because the file is uploaded to LakeFS already (checksums match). This is useful when we work with large files to reduce the amount of network traffic. Nonetheless, in this specific situation the `PUT` is not executed and neither is our commit hook. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fs.put('./data/weather-2010s.json',  f'{REPO_NAME}/{NEW_BRANCH_NAME}/weather-2010.json')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can circumvent this by disabling checking the checksums on a specific put operation. We do this by passing `precheck=False` to the `PUT` operation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fs.put('./data/weather-2010s.json',  f'{REPO_NAME}/{NEW_BRANCH_NAME}/weather-2010.json', precheck=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's transform the data for our use case. We put the transformation into a function such that we can reuse it later\n",
+    "\n",
+    "In this notebook, we follow a simple toy example to predict whether it is raining at the same time tomorrow given weather data from right now. \n",
+    "\n",
+    "However, we will skip a lot of possible feature engineering etc. in order to focus on the application of LakeFS and LakeFS-spec. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import json \n",
+    "\n",
+    "def transform_json_weather_data(filepath):\n",
+    "    f = open(filepath)\n",
+    "    data = json.load(f)\n",
+    "    f.close()\n",
+    "\n",
+    "    df = pd.DataFrame.from_dict(data[\"hourly\"])\n",
+    "    df.time = pd.to_datetime(df.time)\n",
+    "    df['is_raining'] = df.rain > 0\n",
+    "    df['is_raining_in_1_day'] = df.is_raining.shift(24)\n",
+    "    df = df.dropna()\n",
+    "    return df\n",
+    "    \n",
+    "df = transform_json_weather_data('./data/weather-2010s.json')\n",
+    "df.head(5)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us now save this data as a csv into the main branch. The `.to_csv` method calls a `put` operation behind the scenes, our commit hook is called and the file committed. You can verify the saving worked in your LakeFS GUI in the browser when looking at the Comits and uncommitted changes tabs of the main branch. "
```

Comment:
This is probably outdated on `main` - the corresponding file event is `FSEvent.FILEUPLOAD` (also, pandas calls `fs.open()` and not `fs.put`).

```suggestion
    "Now we save this data as a CSV file into the main branch. The `DataFrame.to_csv` method calls an `open` operation behind the scenes, our commit hook is called and the file is committed. You can verify the saving worked in the LakeFS UI in your browser by switching to the commits tab of the main branch. "
```